web-console/knikubevirt/Create-vm/step-2-networking/create-vm-networking.md

```diff
@@ -3,17 +3,20 @@
 ## Step 2: Networking
 
 ![Default](img/3-00.png)
-The user enters the Networking step with an existing NIC pre-defined, set to Type (local). In this step, the 'Review and Create' button is still available to skip the rest of the steps. At this point, the user can select to customize this NIC, add new ones, or remove any they wish.
+The user enters the Networking step with an existing  Network interface pre-defined, set to Type (local). In this step, the 'Review and Create' button is still available to skip the rest of the steps. At this point, the user can select to customize this Network interface, add new ones, or remove any they wish.
 
 ![Adding NIC](img/3-01.png)
-Clicking on 'Add NIC' will pop up a "Create NIC" modal. The NIC name is prepopulated with a numeric count form of existing NICs on this flow. The user may change the NIC name. All the fields are preselected except Type. In order for the user to apply and add this NIC, the user must select one of the type options.
+Clicking on 'Add  Network interface' will pop up a "Add Network interface" modal. The Network interface name is prepopulated with a numeric count form of existing Network interfaces on this flow but the user may change it. In order for the user to apply and add this Network interface, they must select one of the type options.
+In case the user selects 'Masquerade' on the Type field, they can add multiple Ports.
+In that case the user will get a list of Ports (port name, port number, protocol) which will be then forwarded and exposed to the virtual machine. The ports will be displayed in rows so the user can add or remove them.
 
-![Type - Masquerade](img/3-02.png)
-In case the user selects Masquerade on the Type field, new indented fields will appear.
+![masquerade is selected](img/3-03.png)
```

Comment:
We'd only show one row here when they choose to "add port" right? I believe too the (+) should be a (-) so you can remove it. Also did we clarify if when "Masquerade" is chosen there are by default inputs available for at least one port?

Reply:
thanks! I don't think this pattern includes the (+) at all does it?

Reply:
> As far as I remember, I don't think we have discussed what default inputs are available for a port when "Masquerade" is chosen. WDYT? Thanks!
> […](#)
> On Tue, Sep 3, 2019 at 5:27 PM matthew carleton ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In web-console/knikubevirt/Create-vm/step-2-networking/create-vm-networking.md <[#257 (comment)](https://github.com/openshift/openshift-origin-design/pull/257#discussion_r320302306)> : > -![Type - Masquerade](img/3-02.png) -In case the user selects Masquerade on the Type field, new indented fields will appear. +![masquerade is selected](img/3-03.png) We'd only show one row here when they choose to "add port" right? I believe too the (+) should be a (-) so you can remove it. Also did we clarify if when "Masquerade" is chosen there are by default inputs available for at least one port? — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub <#257?email_source=notifications&email_token=AKUDFDB4EUFBZJN5WBBQUDLQHZX3LA5CNFSM4IL7E4HKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCDPMH3Q#pullrequestreview-283034606>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AKUDFDC6TXCRI4YG6LMOKSTQHZX3LANCNFSM4IL7E4HA> .

@suomiy in the case of masquerade is at least one port #, name, and protocol required? If so, I'd say we provide these fields by default when they choose masquerade.

Reply:
I would say they are recommended, not required. Also I do not think it is reasonable to put defaults for these 3 values, because it could confuse user.

Reply:
ok, so we have them choose Masquerade and then choose "Add port" which shows the three blank inputs.

Reply:
Hi @matthewcarleton and @yfrimanm I noticed that the "-" icon is styled a bit differently here than other places I've seen in our PRs. Is this something we would want to standardize in our wireframes? https://github.com/openshift/openshift-origin-design/pull/260/files

Reply:
Thanks, updated.